connector/connect/src/main/protobuf/spark/connect/expressions.proto

```diff
@@ -173,4 +174,18 @@ message Expression {
     // (Optional) Alias metadata expressed as a JSON map.
     optional string metadata = 3;
   }
+
+  // Case statements of the form "CASE WHEN a THEN b [WHEN c THEN d]* [ELSE e] END".
+  message CaseWhen {
```

Comment:
Why not make this an unresolved_function? All it takes is that we add a constructor to CaseWhen.

Reply:
Alternatively you can rewrite this into a series of ifs.

Reply:
hmm for `unresolved_function`, how to deal with the optional `else`?

Reply:
If the the number of expressions is uneven, the last expression is the else expression.

Reply:
If we use this way then have to document it clear for even/odd assumption on the server side.

Reply:
Yeah we do. I am just trying to see if we can keep the number of expressions in the proto to a bare minimum. In this case you can get away with it because it sort of makes sense.

Reply:
I am not sure whether unresolved_function  `when`  can support `otherwise`, let me take a deeper look